src/main/java/edu/cornell/kfs/pdp/businessobject/lookup/CuPayeeACHAccountLookupableHelperServiceImpl.java

```diff
@@ -0,0 +1,162 @@
+package edu.cornell.kfs.pdp.businessobject.lookup;
+
+import java.security.GeneralSecurityException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.lang.StringUtils;
+import org.kuali.kfs.pdp.PdpConstants.PayeeIdTypeCodes;
+import org.kuali.kfs.pdp.PdpPropertyConstants;
+import org.kuali.kfs.pdp.businessobject.PayeeACHAccount;
+import org.kuali.kfs.pdp.businessobject.lookup.PayeeACHAccountLookupableHelperServiceImpl;
+import org.kuali.rice.core.api.criteria.CriteriaLookupService;
+import org.kuali.rice.core.api.criteria.Predicate;
+import org.kuali.rice.core.api.criteria.PredicateFactory;
+import org.kuali.rice.core.api.criteria.PredicateUtils;
+import org.kuali.rice.core.api.criteria.QueryByCriteria;
+import org.kuali.rice.core.api.encryption.EncryptionService;
+import org.kuali.rice.kim.api.identity.Person;
+import org.kuali.rice.kim.api.identity.PersonService;
+import org.kuali.rice.kim.impl.KIMPropertyConstants;
+import org.kuali.rice.kns.lookup.LookupUtils;
+import org.kuali.rice.krad.bo.BusinessObject;
+import org.kuali.rice.krad.util.BeanPropertyComparator;
+import org.kuali.rice.krad.util.KRADConstants;
+
+import edu.cornell.kfs.pdp.CUPdpPropertyConstants;
+
+/**
+ * Custom override of PayeeACHAccountLookupableHelperServiceImpl that performs special handling
+ * when the new principal name search field is given a value.
+ */
+@SuppressWarnings("deprecation")
+public class CuPayeeACHAccountLookupableHelperServiceImpl extends PayeeACHAccountLookupableHelperServiceImpl {
+    private static final long serialVersionUID = 9164050782113012442L;
+
+    private PersonService personService;
+    private CriteriaLookupService criteriaLookupService;
+
+    /**
+     * Overridden to perform custom searching when a principal name is specified on the search screen.
+     * 
+     * @see org.kuali.rice.kns.lookup.KualiLookupableHelperServiceImpl#getSearchResultsHelper(java.util.Map, boolean)
+     */
+    @SuppressWarnings("unchecked")

```

Comment:
Suppressing "unchecked" warnings here is not ideal, but was needed because of this method's reliance on base Rice KNS code that does not have consistent use of generics.